packages/protocol/contracts/layer1/core/impl/CodecSimple.sol

```diff
@@ -140,23 +140,11 @@ contract CodecSimple is ICodec {
     }
 
     /// @inheritdoc ICodec
-    function hashTransitionRecord(IInbox.TransitionRecord calldata _transitionRecord)
```

Comment:
How about we merge Inbox and InboxOptimized into a single Inbox.sol file, then remove CodecSimple and LibHashSimple.